services/iam/src/routes/general.js

```diff
@@ -54,6 +54,49 @@ router.post('/login', authMiddleware.authenticate, authMiddleware.accountIsEnabl
 
 });
 
+router.post('/register', async (req, res, next) => {
+    const tenantName = req.body.companyName;
+    const { username ,firstname,password,lastname} = req.body;
+    let tenantId = ""
+    try {
+        const doc = await TenantDAO.findOne({ name: tenantName });
+        if(doc) {
+            return next({message: "Tenant already exists",status: 409})
```

Comment:
Not positive from documentation if 409 is correct here, but can go with it for now.